caproto/_broadcaster.py

```diff
@@ -97,16 +104,19 @@ def recv(self, byteslike, address):
         -------
         commands : list
         """
-        self.log.debug("Received datagram from %s:%d with %d bytes.",
-                       *address, len(byteslike))
         try:
             commands = read_datagram(byteslike, address, self.their_role)
         except Exception as ex:
             raise RemoteProtocolError(f'Broadcaster malformed packet received:'
                                       f' {ex.__class__.__name__} {ex}') from ex
 
         for command in commands:
-            self.log.debug("%d bytes -> %r", len(command), command)
+            if isinstance(command, Beacon):
+                log = logging.LoggerAdapter(self.beacon_log, {'address': address[0] + ':'+ str(address[1]), 'role': self.role})
```

Comment:
For the sake of speed, I think you can leave it as `{'address': address}` here. Formatters that want to render the address on the screen can pull it apart `'%s:%d' % record.address`. But some handlers (like elastic) will be just as happy or happier with the tuple representation.

Also, this is a situation where I think we can do:

```py
self.beacon_log("%s:%d (%dB) -> %r", *address, len(command), command, extra={...})
```

and avoid the overhead of instantiating a `LoggerAdapter` that we are only using once.

Reply:
To elaborate a little, parsing strings as in `address[0] + ':' + address[1]` takes a bit of time, and this is in a "hot loop" that strongly affects overall performance. So if that string-parsing only needs to be done _some_ of the time but not all of the time, we should defer to later if possible and avoid doing unneeded work.

Reply:
It wasn't clear for me to either choose tuple or str at the beginning. The reason convince me choose str is when user put address for search, there might be a nested search depend on whether there port was given or not. The current AddressFilter allow user give exact address(host:port) or just host. If only host was given, it return True for all address with that host.

Reply:
I can see that reasoning, but I'm not _fully_ convinced that's a sufficiently strong reason to break our pattern of representing addresses as tuples internally, as we do everywhere else in caproto. What would the `AddressFilter` code look like if `record.address` were a tuple?

Reply:
It will be coded like compare `host` and `port` separated. If we still want user input `address` as a str in `Filter`. There will be split code for reconstruct tuple. I understand the addresses are always tuples in Caproto. However, this log is kind of output whether is `tuple` or `str`, it may never need to come back into caproto.

Reply:
I agree with Dan that this is better left as a tuple.  That way in the filter we don't have to split every message (but can instead split the user input) and if we throw this into a database we don't have to split it there as well.